src/plugins/sync-plugin.js

```diff
@@ -794,6 +810,25 @@ const equalAttrs = (pattrs, yattrs) => {
   return eq
 }
 
+const equalMarks = (yattrs, pmarks) => {
+  const keys = Object.keys(yattrs).filter((key) => key.includes('yelement_mark_'));
+
+  let eq =
+    keys.length === pmarks.length;
+
+  let pMarkAttr = nodeMarksToAttributes(pmarks);
+
+  console.log('yattrs:',yattrs , 'pMarkAttr',pMarkAttr);
+
+  for (let i = 0; i < keys.length && eq; i++) {
+    const key = keys[i];
+    const l = JSON.stringify(pMarkAttr[key])
+    const r = yattrs[key]
+    eq = key === 'ychange' || l === r ;
```

Comment:
Once `l` and `r` are objects. You could use `f.equalityDeep` to compare them. Where `import * as f from 'lib0/function'`